### PR TITLE
Query history: Create API to star and unstar query in query history

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

This PR is the first part of moving query history to be stored in database instead of browser's local storage. The design document is [here](https://docs.google.com/document/d/1BdVtoXCxhH0Q6C-O_Z7jKi-u1m5pDoS9_HQKhYu_hEE/edit#)

To make PRs smaller and easier to review, I am splitting it into chunks. Here is the issue with steps: https://github.com/grafana/grafana/issues/44327

This PR:

1. Creates query_history_star table with unique query_uid + user_id index
2. Creates API starring query in query history
 - we always check if query exists in query_history table so only existing queries can be starred
 - we have  unique query_uid + user_id index which means that we can't create duplicate rows in table
3. Creates API for unstarring query in query history
4. Updates `deleteHandler` so we always unstar query before we delete it.

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/grafana/issues/44327

**Special notes for your reviewer**:
1. enable query history `enabled = true` flag in custom.ini

 In https://github.com/grafana/grafana/blob/main/public/app/features/explore/state/query.ts#L324 add:
 ` import { getBackendSrv } from '@grafana/runtime`
 
```
  //To test adding and deleting of query history
  const dataSourceUid = queries[0]?.datasource?.uid;
  if (dataSourceUid) {
    const { result } = await getBackendSrv().post(`/api/query-history`, {
      dataSourceUid,
      queries,
    });

    //Right away we star item and unstar item
    getBackendSrv().post(`/api/query-history/star/${result.uid}`);
    // getBackendSrv().delete(`/api/query-history/star/${result.uid}`);
    // getBackendSrv().delete(`/api/query-history/${result.uid}`);
  }
  // End of test
```

If the feature is enabled, it should be able to star/unstar/delete queries.

This PR is missing adding of API endpoints to the the OpenAPI spec. But after this PR there is only 'getQueryHistory` PR left and I was thinking about adding query history API then. If it is okay with reviewer.

### CODE REVIEW

File: pkg/services/queryhistory/api.go [hunk 1]
```diff
@@ -52,3 +61,37 @@ func (s *QueryHistoryService) deleteHandler(c *models.ReqContext) response.Respo
...
+	if len(queryUID) == 0 || !util.IsValidShortUID(queryUID) {
+		return response.Error(http.StatusNotFound, "Query in query history not found", nil)
+	}
+
+	id, err := s.StarQueryInQueryHistory(c.Req.Context(), c.SignedInUser, queryUID)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to star query in query history", err)
+	}
+
+	return response.JSON(http.StatusOK, StarQueryInQueryHistoryResponse{
```
Comment (ifrost) [inline pkg/services/queryhistory/api.go:76]: Could we return here a DTO as in `patchQueryComment`? We need to get the DTO from the table anyway before starring so it shouldn't be a problem. Getting full DTO could help keeping UI and DB in sync especially once multiple users edit the same query.

**Review by ifrost — COMMENTED** [review]
Should documentation be added in this PR?

File: pkg/services/queryhistory/api.go (hunk 1 above)
Comment (ivanahuckova) [reply pkg/services/queryhistory/api.go:76]: Yes we can! I was thinking about it, but decided not to do that. But will do!

**Comment by ivanahuckova** [comment]
>Should documentation be added in this PR?

Yes! I totally missed that. Good catch.

**Comment by ivanahuckova** [comment]
@papagian Whenever you have time could you plese look at this one. This PR is also missing adding of API endpoints to the the OpenAPI spec. But after this PR there is only 'getQueryHistory` PR left and I was thinking about adding query history API then. If it is okay.

File: docs/sources/http_api/query_history.md [hunk 2]
```diff
@@ -165,3 +164,99 @@ Status codes:
...
+## Star query in Query history
+
+`POST /api/query-history/star/:uid`
+
+Stars query in query history.
+
+**Example request:**
+
+```http
+POST /api/query-history HTTP/1.1
```
Comment (ifrost) [inline docs/sources/http_api/query_history.md:177]:
```suggestion
POST /api/query-history/star/P8zM2I1nz HTTP/1.1
```

File: docs/sources/http_api/query_history.md [hunk 3]
```diff
@@ -165,3 +164,99 @@ Status codes:
...
+## Unstar query in Query history
+
+`DELETE /api/query-history/star/:uid`
+
+Removes stars from query in query history.
+
+**Example request:**
+
+```http
+DELETE /api/query-history HTTP/1.1
```
Comment (ifrost) [inline docs/sources/http_api/query_history.md:225]:
```suggestion
DELETE /api/query-history/star/P8zM2I1nz  HTTP/1.1
```

**Review by ifrost — APPROVED** [review]
Left two small comments regarding docs. Nice work 🚀

**Review by idafurjes — APPROVED** [review]
LGTM

**Pull request merged**